jax/_src/pallas/pallas_call.py

```diff
@@ -1086,6 +1086,27 @@ def wrapped(*args):
       raise ValueError(
           "The kernel function in a pallas_call should return None. "
           f"Found a PyTree: {f_out_tree}")
+    for i_idx, o_idx in input_output_aliases.items():
+      if not (0 <= i_idx < len(flat_in_avals)):
```

Comment:
FYI a neat way to write this is 

```
if i_idx not in range(len(flat_in_avals)):
  ...
```

which matches the wording in the error message :)

Reply:
True, but it is too inefficient to materialize the range and check one by one.

Reply:
`range` doesn't materialize a `list` in 3.X, it just creates a `range` object, but you're right -- there is still an extra allocation.

Reply:
I did not know that `range` objects support O(1) element lookup. I am applying the suggestion.